src/kani-compiler/rustc_codegen_kani/src/codegen/statement.rs

```diff
@@ -78,14 +79,24 @@ impl<'tcx> GotocCtx<'tcx> {
                     if *expected { r } else { Expr::not(r) }
                 };
 
+                let msg = if let AssertKind::BoundsCheck { .. } = msg {
+                    // For bounds check the following panic message is generated at runtime:
+                    // "index out of bounds: the length is {len} but the index is {index}",
+                    // but CBMC only accepts static messages so we don't add values to the message.
+                    "index out of bounds: the length is less than or equal to the given index"
+                } else {
+                    // For all other assert kind we can get the static message.
+                    msg.description()
```

Comment:
I suppose this is where the "attempt to add with overflow" message is coming from?

Reply:
I think it would be worth revisiting the wording of those messages at some point (including the assert messages), and provide a more meaningful wording. For instance, for this one, I would phrase it as "No overflow with add" or "Add operation is free from overflow", or even something more verbose as "Checking for the possibility of an overflow in the add operation". IMHO, such wording would be more suitable for attaching "SUCCESS" or "FAILURE" to.

Reply:
I'm not sure I agree. These wordings are compatible with the wordings a user would get in a concrete run of their code.

Reply:
Yes, but as you say, these wordings are suited for when an assertion fails in a "concrete run". They indicate that this assertion failed in a particular way. The wordings are not suitable for saying "Passed" or "Success" with them. Just my two cents...

Reply:
As discussed offline, we should change how we present the results instead of changing the check message.